src/storage/src/hummock/iterator/forward_user.rs

```diff
@@ -235,7 +235,7 @@ impl<I: HummockIterator<Direction = Forward>> UserIterator<I> {
 
         let full_key = FullKey {
             user_key,
-            epoch: self.read_epoch,
+            epoch_with_gap: EpochWithGap::new_from_epoch(self.read_epoch),
```

Comment:
We should use `EpochWithGap::new(self.read_epoch, <MAX SPILL OFFSEt>)` to do the seek. Otherwise, the latest value written by spill will be skipped.